pkg/api/v20230701preview/openshiftcluster_convert.go

```diff
@@ -54,6 +50,14 @@ func (c openShiftClusterConverter) ToExternal(oc *api.OpenShiftCluster) interfac
 		},
 	}
 
+	if oc.Properties.ServicePrincipalProfile != nil {
+		out.Properties.ServicePrincipalProfile = &ServicePrincipalProfile{}
```

Comment:
Is this assignment necessary? `out.Properties.ServicePrincipalProfile` is getting reassigned to a new struct in the next line.